### PR TITLE
Add withdraw and increaseUnlockTime

### LINKED ISSUE #1474: Add withdraw & increaseUnlockTime actions on ve-hemi
### Feature or change description

Let’s add two missing actions to the ve-hemi actions pkg so the UI can fully manage a lock’s lifecycle. We need a withdraw (burns NFT and returns HEMI after expiry) and an increaseUnlockTime (extends the lock end, rounded to 6-day ticks).

### PR DESCRIPTION
**Pull request by ArturDolzan**
### Description

This PR introduces two new actions, `withdraw` and `increaseUnlockTime`, extending the available interactions with the veHemi contract. Alongside these additions, existing actions were refactored to improve structure and maintainability. Unit tests were also expanded to cover the new flows and validate edge cases.

### Screenshots

No UI changes.

### Related issue(s)

Closes #1474 

### Checklist

- [x] Manual testing passed.
- [x] Automated tests added, or N/A.
- [ ] Documentation updated, or N/A.
- [ ] Environment variables set in CI, or N/A.

### CODE REVIEW

File: packages/ve-hemi-actions/utils.ts [hunk 1]
```diff
@@ -28,6 +18,27 @@ const validateCommonInputs = function ({
...
+  return undefined
+}
+
+const validateTokenId = function (tokenId: bigint) {
+  if (tokenId <= BigInt(0)) {
+    return 'invalid token ID'
+  }
+
+  return undefined
+}
```
[the remark this answers is not in the document]
Comment (ArturDolzan) [reply packages/ve-hemi-actions/utils.ts:30]: That's ok.

File: packages/ve-hemi-actions/actions/wallet/createLock.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    const tokenBalance = await memoizedGetHemiTokenAddress(walletClient).then(
+      hemiTokenAddress =>
+        // Using @ts-expect-error fails to compile so I need to use @ts-ignore
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore because it works on IDE, and when building on its own, but fails when compiling from the portal through next
+        getErc20TokenBalance(walletClient, {
+          account,
+          address: hemiTokenAddress,
+        }),
+    )
```
[the remark this answers is not in the document]
Comment (gndelia) [reply packages/ve-hemi-actions/actions/wallet/createLock.ts:59]: I disagree a bit. I think it's worth using .then in some specific situation to avoid creating variables that are available to the whole async function - this keeps them scoped, without moving the whole function into a nested function level. 

I believe this brings the best of both worlds

I wouldn't use it for large chainings though, but it has its usage here and there.

I agree with the usage of .catch()

File: packages/ve-hemi-actions/test/actions/wallet/increaseAmount.test.ts [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+    vi.mocked(getErc20TokenBalance).mockResolvedValue(BigInt(1000))
+    vi.mocked(getErc20TokenAllowance).mockResolvedValue(BigInt(0))
+    vi.mocked(approveErc20Token).mockResolvedValue(zeroHash)
+    vi.mocked(waitForTransactionReceipt)
+      .mockResolvedValueOnce({
+        status: 'reverted',
+      })
+      .mockResolvedValue({
+        status: 'success',
+      })
```
Comment (gndelia) [inline packages/ve-hemi-actions/test/actions/wallet/increaseAmount.test.ts:312]: If the approve tx fails, I think no other tx should take place 🤔 

```suggestion
    vi.mocked(waitForTransactionReceipt)
      .mockResolvedValueOnce({
        status: 'reverted',
      })
```

File: packages/ve-hemi-actions/test/actions/wallet/increaseUnlockTime.test.ts [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+const validParameters = {
+  account: '0x1234567890123456789012345678901234567890' as const,
+  lockDurationInSeconds: 30 * 24 * 60 * 60, // 30 days
+  tokenId: BigInt(1),
+  walletClient: { chain: hemiSepolia },
+}
+
+describe('increaseUnlockTime', function () {
+  beforeEach(function () {
+    vi.clearAllMocks()
```
Comment (gndelia) [inline packages/ve-hemi-actions/test/actions/wallet/increaseUnlockTime.test.ts:34]:
```suggestion
```
☝🏽  can be removed as it's already done per the [vitest config](https://github.com/hemilabs/ui-monorepo/blob/f2556584a23c76c05b636d7770f6b2d6432b9a15/packages/ve-hemi-actions/vitest.config.ts#L5)

File: packages/ve-hemi-actions/test/actions/wallet/increaseAmount.test.ts [hunk 5]
```diff
@@ -0,0 +1,511 @@
...
+const validParameters = {
+  account: '0x1234567890123456789012345678901234567890' as const,
+  additionalAmount: BigInt(100),
+  tokenId: BigInt(1),
+  walletClient: { chain: hemiSepolia },
+}
+
+describe('increaseAmount', function () {
+  beforeEach(function () {
+    vi.clearAllMocks()
```
Comment (gndelia) [inline packages/ve-hemi-actions/test/actions/wallet/increaseAmount.test.ts:45]:
```suggestion
```

Same as https://github.com/hemilabs/ui-monorepo/pull/1475/files#r2325077440

File: packages/ve-hemi-actions/test/actions/wallet/withdraw.test.ts [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+      amount: BigInt(500),
+      end: pastTimestamp,
+    })
+  })
+
+  it('should emit "withdraw-failed-validation" if the account is not a valid address', async function () {
+    const { emitter, promise } = withdraw({
+      ...validParameters,
+      account: 'invalid-address',
+      walletClient: { chain: hemiSepolia },
```
Comment (gndelia) [inline packages/ve-hemi-actions/test/actions/wallet/withdraw.test.ts:49]: Same as the other file

```suggestion
```

(And in the following tests too)

File: packages/ve-hemi-actions/actions/wallet/createLock.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ArturDolzan) [reply packages/ve-hemi-actions/actions/wallet/createLock.ts]: Applied here 1cc5f78bf8bac889a2129625fdf46b34644c3ca1. Thanks!

File: packages/ve-hemi-actions/actions/wallet/withdraw.ts [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+    const { amount, end } = await getLockedBalance(walletClient, tokenId)
+    const now = BigInt(Math.floor(Date.now() / 1000))
+
+    if (amount <= 0) {
+      return { canWithdraw: false, reason: 'no existing lock' }
+    }
+
+    if (end > now) {
+      return { canWithdraw: false, reason: 'lock not yet expired' }
+    }
```
[the remark this answers is not in the document]
Comment (ArturDolzan) [reply packages/ve-hemi-actions/actions/wallet/withdraw.ts:58]: Applied here 1cc5f78bf8bac889a2129625fdf46b34644c3ca1. Thanks!

File: packages/ve-hemi-actions/test/actions/wallet/increaseAmount.test.ts (hunk 3 above)
Comment (ArturDolzan) [reply packages/ve-hemi-actions/test/actions/wallet/increaseAmount.test.ts]: Fixed here 1cc5f78bf8bac889a2129625fdf46b34644c3ca1. Thanks!

File: packages/ve-hemi-actions/test/actions/wallet/increaseUnlockTime.test.ts (hunk 4 above)
Comment (ArturDolzan) [reply packages/ve-hemi-actions/test/actions/wallet/increaseUnlockTime.test.ts]: Applied here 1cc5f78bf8bac889a2129625fdf46b34644c3ca1. Thanks!

File: packages/ve-hemi-actions/test/actions/wallet/increaseAmount.test.ts (hunk 5 above)
Comment (ArturDolzan) [reply packages/ve-hemi-actions/test/actions/wallet/increaseAmount.test.ts]: Applied here 1cc5f78bf8bac889a2129625fdf46b34644c3ca1. Thanks!

File: packages/ve-hemi-actions/test/actions/wallet/increaseUnlockTime.test.ts [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+    expect(increaseUnlockTimeFailedValidation).toHaveBeenCalledExactlyOnceWith(
+      'account is not a valid address',
+    )
+  })
+
+  it('should emit "increase-unlock-time-failed-validation" if tokenId is zero', async function () {
+    const { emitter, promise } = increaseUnlockTime({
+      ...validParameters,
+      tokenId: BigInt(0),
+      walletClient: { chain: hemiSepolia },
```
[the remark this answers is not in the document]
Comment (ArturDolzan) [reply packages/ve-hemi-actions/test/actions/wallet/increaseUnlockTime.test.ts]: Applied here 1cc5f78bf8bac889a2129625fdf46b34644c3ca1. Thanks!

File: packages/ve-hemi-actions/test/actions/wallet/withdraw.test.ts (hunk 6 above)
Comment (ArturDolzan) [reply packages/ve-hemi-actions/test/actions/wallet/withdraw.test.ts]: Applied here 1cc5f78bf8bac889a2129625fdf46b34644c3ca1. Thanks!

File: packages/ve-hemi-actions/test/utils.test.ts [hunk 9]
```diff
@@ -252,3 +254,172 @@ describe('validateIncreaseAmountInputs', function () {
...
+
+  it('should accept valid token ID', function () {
+    expect(
+      validateWithdrawInputs({
+        ...validParams,
+        tokenId: BigInt(123),
+      }),
+    ).toBeUndefined()
+  })
+})
```
[the remark this answers is not in the document]
Comment (ArturDolzan) [reply packages/ve-hemi-actions/test/utils.test.ts]: Yep, I think Claude and I over tested a bit 😅. Improved here 1cc5f78bf8bac889a2129625fdf46b34644c3ca1. Thanks!

**Review by arielmergen — APPROVED** [review]
